### PR TITLE
EX-1854 Add Ambient for AEVM

### PR DESCRIPTION
**Pull request by phqb**

### CODE REVIEW

File: pkg/liquidity-source/ambient/types.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	LastCreateTime uint64 `json:"lastCreateTime"`
+}
+
+type StaticExtra struct {
+	// Base token, 0x0 if native token
+	Base string `json:"base"`
+	// Quote token
+	Quote string `json:"quote"`
+	// The index or discriminator of the pool in a logical list of pools with (base, quote) tokens. Default is "420".
+	PoolIdx string `json:"pool_idx"`
```
Comment (piavgh) [inline pkg/liquidity-source/ambient/types.go:26]: we should use camelCase here, same with SwapAddress below

File: pkg/liquidity-source/ambient/types.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package ambient
+
+type FetchPoolsResponse struct {
+	Pools []Pool `json:"pools"`
+}
+
+type Pool struct {
+	ID          string `json:"id"`
+	BlockCreate string `json:"blockCreate"`
+	TimeCreate  uint64 `json:"timeCreate,string"`
```
Comment (piavgh) [inline pkg/liquidity-source/ambient/types.go:10]: what is the purpose of the `,string` here?

File: pkg/liquidity-source/ambient/types.go (hunk 1 above)
Comment (phqb) [reply pkg/liquidity-source/ambient/types.go]: done

File: pkg/liquidity-source/ambient/types.go (hunk 2 above)
Comment (phqb) [reply pkg/liquidity-source/ambient/types.go:10]: It tells json to unmarshal string as number, for example `{"timeCreate":"123"}` will be unmarshaled as `Pool{ TimeCreate: 123 }`.

**Review by piavgh — APPROVED** [review]

**Pull request merged**